src/client/tab.rs

```diff
@@ -1889,6 +1904,9 @@ impl Tab {
         } else {
             self.active_terminal = Some(active_terminal.unwrap().pid());
         }
+        self.send_pty_instructions
+            .send(PtyInstruction::UpdateActivePane(self.active_terminal))
+            .unwrap();
```

Comment:
I think this is the reason some of the tests are failing. Be sure to only send this instruction if the `active_terminal` has actually changed (eg. here move this inside the `else` block above).